mitiq/zne/inference.py

```diff
@@ -454,38 +459,20 @@ def _batch_populate_instack(self) -> None:
                 {"scale_factor": scale} for scale in self._scale_factors
             ]
 
-    def _get_keyword_args(self) -> List[Dict[str, Any]]:
+    def _get_keyword_args(self) -> Union[List[Dict[str, Any]], None]:
+        """Returns a list of keyword dictionaries to be used by a batched
+        executor. If keywords are not necessary, None is returned.
+        """
+
         params = deepcopy(self._instack)
         for d in params:
             _ = d.pop("scale_factor")
-        return params
 
-    def next(self) -> Dict[str, float]:
-        """Returns a dictionary of parameters to execute a circuit at."""
-        in_params = {}
-        try:
-            index = len(self._outstack)
-            in_params["scale_factor"] = self._scale_factors[index]
-            if self._shot_list:
-                in_params["shots"] = self._shot_list[index]
-        except IndexError:
-            raise IndexError(
-                "BatchedFactory cannot take another step. "
-                "Number of batched scale_factors"
-                f" ({len(self._scale_factors)}) exceeded."
-            )
-        return in_params
+        # Check if keywords are absent
+        if params is [{} for _ in params]:
+            return None
```

Comment:
Why not return the list of empty dictionaries? This is perfectly fine and gets rid of a check for keyword args in line 395.

Reply:
Thanks, good point.